src/equisolve/vanilla/preprocessing/_base.py

```diff
@@ -0,0 +1,158 @@
+        Must be an object or a list of objects that contain the properties.
+        Currenly supported are ase.Atoms objects.
+
+    :param property: string or function that returns property
+        The string is used to extract the property out of the frame.
+        The way the string is used depends on the frame type:
+
+        ase.Atoms
+        ```python
+            frame_property = frame.info[property]
+        ```
+
+        If a function is given, the function is used to extract the property out
+        of the frame
+        ```python
+            frame_property = property(frame)
+        ```
+        chemfiles.Frame
+        TODO
+
+    :param positions_gradient: string or function that returns array of shape (len(frame), 3)
+        The string is used to extract the gradient of the property wrt. positions out of the frame.
+        The way the string is used depends on the frame type:
+
+        ase.Atoms
+        ```python
+            property_positions_gradient = frame.arrays[positions_gradient]
+        ```
+        chemfiles.Frame
+        TODO
+
+        If a function is given, the function is used to extract the the gradient of the property wrt. positions out
+        of the frame
+        ```python
+            property_positions_gradient = positions_gradient(frame)
+        ```
+
+    :param cell_gradient: string or function that returns array of shape (3, 3)
+        NOT IMPLEMENTED
+
+    """
+
+    if not isinstance(frames, list):
+        frames = [frames]
+    if isinstance(frames[0], ase.Atoms):
+        return _ase_frames_property_to_tensormap(frames, property, positions_gradient, cell_gradient, is_structure_property)
+    elif isinstance(frames[0], chemfiles.Frame):
+        raise NotImplementedError("Extracting porperties out of frame objects of type chemfiles.Frame has not been implemented yet.")
+    else:
+        raise NotImplementedError(f"Extracting porperties out of frame objects of type {type(frames[0])} is not supported.")
+
+
+def _ase_frames_property_to_tensormap(frames, property=None, positions_gradient=None, cell_gradient=None, is_structure_property=True):
+    if cell_gradient is not None:
+        raise NotImplementedError("cell_gradient is not supported at the moment.")
+    if not(is_structure_property):
+        raise NotImplementedError("Support for environment properties has not been implemented yet.")
+
+    if isinstance(property, str):
+        get_property = lambda frame : frame.info[property]
+    else:
+        get_property = property
+    if isinstance(positions_gradient, str):
+        get_positions_gradient = _get_default_positions_gradient_extraction(frames[0], positions_gradient)
+    else:
+        get_positions_gradient = positions_gradient
+    #if isinstance(cell_gradient, str):
+    #    get_stress = lambda frame : -frame.arrays[property]
+
+
+    keys = Labels(
+        names=["idx"],
+        values=np.array([[0]], dtype=np.int32)
+    )
```

Comment:
there is a function for when you just need a Labels with a single entry:

```suggestion
    keys = Labels.single()
```